src/blocks/CaseStudyParallax/index.tsx

```diff
@@ -0,0 +1,307 @@
+          <Button
+            label={'Read the case study'}
+            hideHorizontalBorders
+            appearance={'default'}
+            icon="arrow"
+            className={classes.caseStudyButton}
+            href={`/case-studies/${card.caseStudy.slug}`}
+            el="a"
+            aria-hidden={!isVisible}
+            disabled={!isVisible}
+          />
+        </div>
+      )}
+    </div>
+  )
+}
+
+export const QuoteStickyBlock: React.FC<StickyBlockProps> = props => {
+  const { caseStudyParallaxFields, currentIndex } = props
+
+  if (caseStudyParallaxFields?.cards && caseStudyParallaxFields?.cards?.length > 0) {
+    return (
+      <div className={[classes.stickyBlock, 'grid'].filter(Boolean).join(' ')}>
+        {caseStudyParallaxFields?.cards.map((card, index) => {
+          const isVisible = index === currentIndex
+
+          return (
+            <QuoteBlock isVisible={isVisible} card={card} className={classes.stickyBlockItem} />
+          )
+        })}
+      </div>
+    )
+  }
+  return null
+}
+
+export const CaseStudyParallax: React.FC<Props> = props => {
+  const { caseStudyParallaxFields } = props
+  const [activeIndex, setActiveIndex] = React.useState<number>(0)
+  const [scrollProgress, setScrollProgress] = React.useState<number>(0)
+  const [delayNavScroll, setDelayNavScroll] = React.useState<boolean>(false)
+  const containerRef = React.useRef<HTMLDivElement>(null)
+  const cardsRef = React.useRef<HTMLDivElement[]>([])
+  const navGridRef = React.useRef<HTMLDivElement>(null)
+  const navButtonsRef = React.useRef<HTMLDivElement[]>([])
+  const id = React.useId()
+  const containerWidth = useResize(containerRef)
+
+  React.useEffect(() => {
+    if (scrollProgress) {
+      let newIndex = 0
+      if (scrollProgress < 25) {
+        newIndex = 0
+      }
+      if (scrollProgress > 25 && scrollProgress < 50) {
+        newIndex = 1
+      }
+      if (scrollProgress > 50 && scrollProgress < 75) {
+        newIndex = 2
+      }
+      if (scrollProgress > 75) {
+        newIndex = 3
+      }
+
+      if (newIndex !== activeIndex) {
+        setActiveIndex(newIndex)
+
+        if (navButtonsRef.current?.length && navGridRef.current) {
+          if (delayNavScroll) {
+            /* This logic is in a timeout so that on mobile scroll() doesnt block the other scrollIntoView function */
+            setTimeout(() => {
+              const target = navButtonsRef.current[newIndex]
+              const offset = target.offsetLeft > 0 ? target.offsetLeft : 0
+              navGridRef.current?.scroll(offset, 0)
+              setDelayNavScroll(false)
+            }, 500)
+          } else {
+            const target = navButtonsRef.current[newIndex]
+            const offset = target.offsetLeft > 0 ? target.offsetLeft : 0
+            navGridRef.current?.scroll(offset, 0)
+          }
+        }
+      }
+    }
+  }, [scrollProgress, navButtonsRef, navGridRef, delayNavScroll])
```

Comment:
@paulpopus Getting a dependency warning in useEffect - missing dependency `activeIndex`